### ISSUE TITLE
WASM exceed issue: Insufficient resources: memory minimum size of 2092 pages exceeds memory limits

### ISSUE THREAD

**Issue by hbulgarini** [SubstrateChess/pallet-chess#27]
When the pallet is added in Trappist, the chain gets stalled and it stops producing blocks. Apparently, the error starts when the validators wants to run PoV function for the parachain:

`2023-05-04 17:25:58 Failed to validate candidate para_id=Id(1836) error=InvalidCandidate(WorkerReportedError("execute: execute error: RuntimeConstruction(Other(\"failed to instantiate a new WASM module instance: Insufficient resources: memory minimum size of 2092 pages exceeds memory limits\"))"))`

Tests were done running release 0.9.37 during the preparations for Trappist deployment on Rococo.

CC: @kalaninja

**Comment by Moliholy** [comment]
As discussed with @hbulgarini, he is able to reproduce the problem, but I am not. Instead, everything works fine and all parachains produce blocks without any issue. These are the wasm sizes I get when compiling trappist:

```
$ ls -liash target/release/wbuild/trappist-runtime/trappist_runtime.*
1787634  2704 -rw-r--r--@ 1 molina  staff   1.3M May 30 17:08 target/release/wbuild/trappist-runtime/trappist_runtime.compact.compressed.wasm
1787600 16232 -rw-r--r--@ 1 molina  staff   7.9M May 30 17:09 target/release/wbuild/trappist-runtime/trappist_runtime.compact.wasm
1787635 15944 -rwxr-xr-x@ 1 molina  staff   7.8M May 30 18:06 target/release/wbuild/trappist-runtime/trappist_runtime.wasm
```

I don't really know which of the three is causing the issue, to be honest. Do you guys have any clue?

**Comment by AlexD10S** [comment]
Any updates with this issue? I want to help with it.
I have been building a [front end app](https://github.com/AlexD10S/chess-substrate-dapp) to interact with the Chess pallet, and it would be great to deploy it if we have the chess pallet live.

**Comment by hbulgarini** [comment]
> Any updates with this issue? I want to help with it. I have been building a [front end app](https://github.com/AlexD10S/chess-substrate-dapp) to interact with the Chess pallet, and it would be great to deploy it if we have the chess pallet live.

Not really, since it was removed from trappist runtime we are not much diving into this project anymore.
